inf-clojure.el

```diff
@@ -725,19 +731,24 @@ The value is nil if it can't find one."
 (defun inf-clojure-show-var-documentation (var)
   "Send a form to the inferior Clojure to give documentation for VAR.
 See function `inf-clojure-var-doc-form'."
-  (interactive (inf-clojure-symprompt "Var doc" (inf-clojure-var-at-pt)))
-  (comint-proc-query (inf-clojure-proc) (format (inf-clojure-var-doc-form) var)))
+  (interactive "P")
+  (let ((var (if (inf-clojure--prompt-on-commands-p var)
+                 (car (inf-clojure-symprompt "Var doc" (inf-clojure-var-at-pt)))
+               (inf-clojure-var-at-pt))))
+    (comint-proc-query (inf-clojure-proc) (format (inf-clojure-var-doc-form) var))))
 
 (defun inf-clojure-show-var-source (var)
-  "Send a form to the inferior Clojure to give source for VAR.
-See variable `inf-clojure-var-source-form'."
-  (interactive (inf-clojure-symprompt "Var source" (inf-clojure-var-at-pt)))
-  (comint-proc-query (inf-clojure-proc) (format inf-clojure-var-source-form var)))
+  "Send a command to the inferior Clojure to give source for VAR.
+See variable `inf-clojure-var-source-command'."
+  (interactive "P")
+  (let ((var (if (inf-clojure--prompt-on-commands-p var)
```

Comment:
One more thing - by moving out the var init from `interactive` you're going to break those functions when invoked without parameters. I'm guessing you didn't really test this code. :-) You'll have to make params like `var` optional and prompt for them only when they are not passed explicitly (or remove them completely).